shm/Cargo.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <string>
+
+class Cargo {
+public:
+    Cargo(const std::string& name, size_t amount, size_t basePrice)
+        : name_(name), amount_(amount), basePrice_(basePrice) {}
```

Comment:
I think it will be better to move constructors/destructors to cpp file

Reply:
I would agree if there was any body to the constructor. As it is it's just assign values.